src/bleaksport/trainer.py

```
@@ -403,20 +414,36 @@ def _on_ftms_event(event: Any) -> None:
         self._last = None
 
+        disconnected = True
+        # pyftms only calls BleakClient.disconnect() while is_connected is
+        # true. Capture the client first so a disconnect callback deleting
+        # machine._cli cannot prevent closure of Bleak's D-Bus connection.
+        client = getattr(m, "_cli", None)
         async with self._ble_lock:
-            with contextlib.suppress(Exception):
+            try:
                 await m.disconnect()
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)